src/libs/memoize/cache/arrayCacheBuilder.ts

```diff
@@ -0,0 +1,76 @@
+import type {Cache, CacheOpts} from '@libs/memoize/types';
+import {getEqualityComparator} from '@libs/memoize/utils';
+
+/**
+ * Builder of the cache using `Array` primitive under the hood.
+ * @param opts - Cache options, check `CacheOpts` type for more details.
+ * @returns
+ */
+function buildArrayCache<K extends unknown[], V>(opts: CacheOpts): Cache<K, V> {
+    const cache: Array<[K, V]> = [];
+
+    const keyComparator = getEqualityComparator(opts);
+
+    function getKeyIndex(key: K) {
+        for (let i = cache.length - 1; i >= 0; i--) {
+            if (keyComparator(cache[i][0], key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    return {
+        get(key) {
+            const index = getKeyIndex(key);
+
+            if (index !== -1) {
+                const [entry] = cache.splice(index, 1);
+                cache.push(entry);
+                return {value: entry[1]};
+            }
+        },
+        set(key, value) {
+            const index = getKeyIndex(key);
+
+            if (index !== -1) {
+                cache.splice(index, 1);
+            }
+
+            cache.push([key, value]);
+
+            if (cache.length > opts.maxSize) {
+                cache.shift();
+            }
+        },
+        delete(key) {
```

Comment:
It's also worth noting that there are a few things we've implemented here but aren't actually using anywhere:

- delete
- clear
- snapshot
- size

Reply:
Agree, YAGNI all the way! The idea behind it was mostly having a stable (following `Map` API) interface for the future, but we can delete (huh :D) `delete` and `clear` to reduce the clutter.

For the rest, `size` is used as a part of monitoring and I feel like it is safe to assume that `snapshot` utilities might be used in monitoring as well in near future. Also they are quite convenient tool for debugging. 

Debugging definitely should not be the first priority - the app size is definitely bigger concern, but it's still tiny tradeoff I guesss. So let me know what you think, thanks!

Reply:
looks like you removed `delete` and `clear`, which is good enough for me for now 👍🏼